### PR TITLE
[feature] Prompt user to manually specify the push remote

### PR DESCRIPTION
**Pull request by a7r3**
Attempts to solve #6 

Additional changes which can be implemented :

* Check if the user-provided URL and Branch are valid (`git ls-remote` can do that, under any valid git repository) before setting it as **origin**

### CODE REVIEW

**Comment by maurobringolf** [comment]
I just want to quickly thank you for doing this! #hacktoberfest 😀 I'll try and review/merge asap.

File: github-release
Comment (maurobringolf) [inline github-release]: This is great, neat prompt! But currently it is not working correctly because the `$REPOSITORY` is not set after adding the new remote. I see a couple options:

* Copy the line that reads the repository handle down into this case after the new remote is added.
* Make the line that reads the repository handle a function and call it in the two places.
* Somehow check if remote origin exists and is a GitHub repo first, ask for it if not. And then extract the repository handle only once after that.

All of them are fine with me 👍

**Review by maurobringolf — CHANGES_REQUESTED** [review]

File: github-release
Comment (a7r3) [reply github-release]: 2nd option is what I thought of doing. Would do now
3rd option, Yes, I specified it could be done on request

Comment (maurobringolf) [reply github-release]: So I'd suggest going with the second option.

File: github-release
Comment (a7r3) [inline github-release]: * I've set the timeout to be 30s, opinions ?
  - I've done this since `ls-remote` may take too much time when the Repo URL isn't valid

* I've chosen the 3rd Option :laughing: 

Oh, and BTW, take a read through the commit message too

Comment (maurobringolf) [reply github-release]: I like the third option 😉 . A couple last things before I merge this:

* `timeout` is not universal, does not exist on MacOS per default for example. I would suggest we leave it up to the user and just read `REPO_URL` and `REPO_BRANCH` once. I want to improve error messages on failed API requests anyway ( #24  ), so this case will be covered there. Basically remove the while loop entirely.

* Currently `if ! git config --get remote.origin.url` prints the result to the console, which is not what we want. Using `[ -z "$(git config --get remote.origin.url)" ]` fixes this for me. I'm not 100% sure about the exact difference between the two, but on MacOS there is no output on the second one, which is more appropriate here.

**Review by maurobringolf — CHANGES_REQUESTED** [review]

File: github-release
Comment (a7r3) [reply github-release]: Ok, so the `ls-remote` part would be removed.
Fine with me, since you said it'd taken care on sending an API request

I'd come up with the **First Option** essentially soon

**Comment by a7r3** [comment]
@maurobringolf Rebased and pushed first commit

* Removed the `while` condition
* Retained the `get_repository_details` functions, felt it might be useful for you later when API requests fail.

**Comment by maurobringolf** [comment]
Awesome, it works as expected. Thank you so much! 🎉

**Comment by a7r3** [comment]
Had a good discussion, thanks!

**Pull request merged**
